src/Adapter/Form/ChoiceProvider/CategoryTreeChoiceProvider.php

```diff
@@ -44,22 +44,29 @@ final class CategoryTreeChoiceProvider implements FormChoiceProviderInterface
      */
     private $contextShopRootCategoryId;
 
+    /**
+     * @var boolean
+     */
+    private $enabledCategoriesOnly;
+
     /**
      * @param CategoryDataProvider $categoryDataProvider
      * @param $contextShopRootCategoryId
+     * @param boolean $enabledCategoriesOnly
      */
-    public function __construct(CategoryDataProvider $categoryDataProvider, $contextShopRootCategoryId)
+    public function __construct(CategoryDataProvider $categoryDataProvider, $contextShopRootCategoryId, $enabledCategoriesOnly = false)
```

Comment:
I think to avoid any misunderstanding if someone uses this again, you should not have a default value
`$enabledCategoriesOnly` should be required, and you then have to update the service definition, this way if someone copies it he can see all the arguments

Reply:
While I agree in principle, adding a new parameter without default value would be a breaking change.

We should update the service definition though, even if it's the same value as the default.

Reply:
done !